ocean-client/src/programs/vault-maxi-program.ts

```diff
@@ -236,6 +237,59 @@ export class VaultMaxiProgram extends CommonProgram {
         return false
     }
 
+    
+    async removeExposure(vault: LoanVaultActive, telegram: Telegram): Promise<boolean> {
+        let pool: PoolPairData = (await this.getPool(this.lmPair))!!
+        const balances = await this.getTokenBalances()
+        const lpTokens = balances.get(this.lmPair)
+        const tokenLoan = vault.loanAmounts.find(loan => loan.symbol == this.settings.LMToken)
+        const dusdLoan = vault.loanAmounts.find(loan => loan.symbol == "DUSD")
+        const stock_per_token = new BigNumber(pool!.tokenA.reserve).div(pool!.totalLiquidity.token)
+        const dusd_per_token = new BigNumber(pool!.tokenB.reserve).div(pool!.totalLiquidity.token)
+        if(!tokenLoan || !dusdLoan || !lpTokens ) {
+            await telegram.send("ERROR: can't withdraw from pool, no tokens left or no loans left")
+            console.error("can't withdraw from pool, no tokens left or no loans left")
+            return false
+        }
+        const maxTokenFromStock= new BigNumber(tokenLoan!.amount).div(stock_per_token)
+        const maxTokenFromDUSD = new BigNumber(dusdLoan!.amount).div(dusd_per_token)
+        const usedTokens= BigNumber.min(lpTokens.amount,maxTokenFromDUSD,maxTokenFromStock)
```

Comment:
Means we are not just blindly removing everything of the configured LM pool.
For example I have 300 TSLA-DUSD, but only about 200 TSLA-DUSD are coming from loans. We are calculating that we don't need to remove all 300, but instead only 200, because the 200 are sufficient enough to payback as much loans as possible?

Reply:
exactly, cause I thought we don't know whats happening on the wallet. users might have put other funds into the LM too. would be a pain if we remove that from LM just to use a small part for paying back loans.